lib/pg_party/model/shared_methods.rb

```diff
@@ -8,7 +8,10 @@ module SharedMethods
       def reset_primary_key
         if self != base_class
           base_class.primary_key
-        elsif partition_name = partitions.first
+        elsif (partitions = partitions(include_subpartitions: true)) && partitions.any?
+          partition_name = partitions.detect { |p| !connection.table_partitioned?(p) }
+          raise 'No child partitions exist for this model' unless partition_name
```

Comment:
Just a note for myself. I think it's time to introduce `PgParty` specific error classes